ethcore/src/engines/mod.rs

```diff
@@ -427,6 +431,9 @@ pub trait Engine<M: Machine>: Sync + Send {
 
 	/// Check whether the given new block is the best block, after finalization check.
 	fn fork_choice(&self, new: &M::ExtendedHeader, best: &M::ExtendedHeader) -> ForkChoice;
+
+	/// Return author should used in executing txns for this block.
+	fn executive_author(&self, header: &M::Header) -> Address { header.author().clone() }
```

Comment:
I'm not if we want this in this trait. 

Also clone-on-copy type

Reply:
why ? this is also how GETH does it.